File: dotnet/DesignPatterns/BehavioralDesignPatterns/ChainOfResponsibilityDesignPattern/ChainOfResponsibilityDesignPattern.Example3/Handlers/Level2SupportHandler.cs
namespace ChainOfResponsibilityDesignPattern.Example3.Handlers;

public class Level2SupportHandler : BaseSupportHandler
{
    public Level2SupportHandler() : base("Level 2 Support") { }

    public override void HandleTicket(SupportTicket ticket)
    {
        // Level 2 can handle Medium priority tickets and Technical/Account issues
        if (CanHandle(ticket, TicketPriority.Medium, new List<TicketCategory> 
            { TicketCategory.Technical, TicketCategory.Account, TicketCategory.General }))
        {
            ticket.Resolve(HandlerName, "Performed advanced diagnostics and applied technical solution.");
        }
        else
        {
            base.HandleTicket(ticket);
        }
    }
}

File: nodejs/DesignPatterns/BehavioralDesignPatterns/ChainOfResponsibilityDesignPattern/ChainOfResponsibilityDesignPattern.Example3/src/Handlers/Level2SupportHandler.ts
import { SupportTicket, TicketCategory, TicketPriority } from "../SupportTicket";
import { BaseSupportHandler } from "../BaseSupportHandler";

export class Level2SupportHandler extends BaseSupportHandler {
  constructor() { super("Level 2 Support"); }

  handleTicket(ticket: SupportTicket): void {
    if (
      this.canHandle(ticket, TicketPriority.Medium, [
        TicketCategory.Technical,
        TicketCategory.Account,
        TicketCategory.General,
      ])
    ) {
      ticket.resolve(this.handlerName, "Performed advanced diagnostics and applied technical solution.");
    } else {
      super.handleTicket(ticket);
    }
  }
}
